project-koku/koku
# Adding tag key API for OCP

**Overview:**
`/api/v1/tags/ocp` will return all tag keys that match a `resolution`, `time_scope_value`, and `time_scope_units` query.

Some refactoring was done so that the tags API can share the same query logic with the reports API.  This involved creating a top-level `QueryHandler` which handles the query criteria mentioned above.  There are certainly more items that could be moved into `QueryHandler` but for this PR I wanted to limit it to mainly what was needed for the time based queries.

**Testing**
1. Ingest AWS and OCP report data.
2. Verify that the OCP and AWS report queries return the same data with `master`
3. Verify that the new /api/v1/tags/ocp/ API returns tag keys as expected.

**Test Results**
[koku_tag_key_api_ut.txt](https://github.com/project-koku/koku/files/2690774/koku_tag_key_api_ut.txt)
